crates/vm/levm/src/precompiles.rs

```diff
@@ -1681,3 +1701,52 @@ fn parse_scalar(scalar_raw_bytes: Option<&[u8]>) -> Result<Scalar, VMError> {
     scalar_le.reverse();
     Ok(Scalar::from_raw(scalar_le))
 }
+
+/// Signature verification in the “secp256r1” elliptic curve
+pub fn p_256_verify(
+    calldata: &Bytes,
+    gas_for_call: u64,
+    consumed_gas: &mut u64,
+) -> Result<Bytes, VMError> {
+    let gas_cost = P256VERIFY_COST;
+
+    increase_precompile_consumed_gas(gas_for_call, gas_cost, consumed_gas)?;
+
+    // If calldata does not reach the required length, we should fill the rest with zeros
+    let calldata = fill_with_zeros(calldata, 160)?;
+
+    // Parse parameters
+    let message_hash = calldata.get(0..32).ok_or(InternalError::SlicingError)?;
+    let r = calldata.get(32..64).ok_or(InternalError::SlicingError)?;
+    let s = calldata.get(64..96).ok_or(InternalError::SlicingError)?;
+    let x = calldata.get(96..128).ok_or(InternalError::SlicingError)?;
+    let y = calldata.get(128..160).ok_or(InternalError::SlicingError)?;
+
+    // Build verifier
+    let Ok(verifier) = VerifyingKey::from_encoded_point(&EncodedPoint::from_affine_coordinates(
+        x.into(),
+        y.into(),
+        false,
+    )) else {
+        return Ok(Bytes::new());
+    };
+
+    // Build signature
+    let r: [u8; 32] = r.try_into().map_err(|_| InternalError::SlicingError)?;
+    let s: [u8; 32] = s.try_into().map_err(|_| InternalError::SlicingError)?;
```

Comment:
These errors are okay as `InternalError::SlicingError` since if we fail here it'd be because of an internal error (the input was validated before, so there must have been a mistake after the first validation if we result in this error).